daemon/redis.c

```diff
@@ -1645,6 +2220,590 @@ static void redis_update_dtls_fingerprint(struct redis *r, const char *pref, con
 		f->hash_func->name,
 		S_LEN(f->digest, sizeof(f->digest)));
 }
+/**
+ * encodes the few (k,v) pairs for one call under one json structure
+ */
+
+char* redis_encode_json(struct call *c) {
+
+	GList *l=0,*k=0, *m=0, *n=0;
+	struct endpoint_map *ep;
+	struct call_media *media;
+	struct rtp_payload_type *pt;
+	struct stream_fd *sfd;
+	struct packet_stream *ps;
+	struct intf_list *il;
+	struct call_monologue *ml, *ml2;
+#if !(GLIB_CHECK_VERSION(2,36,0))
+	g_type_init();
+#endif
+	JsonBuilder *builder = json_builder_new ();
+
+	char tmp[2048]; ZERO(tmp);
+
+	json_builder_begin_object (builder);
+	{
+		sprintf(tmp,"json-%s",STRSTR(&c->callid));
```

Comment:
The appropriate way to use a `str` in a printf-like format is with `STR()` and `STR_FMT`, like this:
`sprintf(tmp,"json-" STRFMT, STR(&c->callid));`
The `STRSTR` macro should probably be removed as there's no guarantee the string is null terminated. (In these cases, the strings actually should all be null terminated, but it's better not to trust this.)